src/mysqlserver.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env python3
+# Copyright 2021 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import logging
+
+from mysql.connector import connect, Error
+
+logger = logging.getLogger(__name__)
+
+
+class MySQL:
+    """MySQL class docstring"""
+
+    def __init__(self, config):
+        self.app_name = config["app_name"]
+        self.host = config["host"]
+        self.port = config["port"]
+        self.user_name = config["user_name"]
+        self.mysql_root_password = config["mysql_root_password"]
+
+    def _get_client(self):
+        """Returns MySQL connection"""
+        connection = None
+        try:
+            connection = connect(
+                host=self.host,
+                user=self.user_name,
+                passwd=self.mysql_root_password,
+            )
+            return connection
+        except Error as e:
+            raise e
+
+    def is_ready(self) -> bool:
+        """Returns if MySQL is up and running"""
+        ready = False
+
+        try:
+            client = self._get_client()
+            ready = True
+            if client is not None:
```

Comment:
Feels a bit strange to be checking if client is not None after setting ready is True .

Reply:
Fixed